util/src/main/java/io/kubernetes/client/util/WebSockets.java

```diff
@@ -81,13 +83,28 @@
      * @param listener The socket listener to handle socket events
      */
     public static void stream(String path, String method, ApiClient client, SocketListener listener) throws ApiException, IOException {
+        stream(path, method, new ArrayList<Pair>(), client, listener);
+    }
+
+    public static void stream(String path, String method, List<Pair> queryParams, ApiClient client, SocketListener listener) throws ApiException, IOException {
+            
         HashMap<String, String> headers = new HashMap<String, String>();
         String allProtocols = String.format("%s,%s,%s,%s", V4_STREAM_PROTOCOL, V3_STREAM_PROTOCOL, V2_STREAM_PROTOCOL, V1_STREAM_PROTOCOL);
         headers.put(STREAM_PROTOCOL_HEADER, allProtocols);
         headers.put(HttpHeaders.CONNECTION, HttpHeaders.UPGRADE);
         headers.put(HttpHeaders.UPGRADE, SPDY_3_1);
 
-        Request request = client.buildRequest(path, method, new ArrayList<Pair>(), new ArrayList<Pair>(), null, headers, new HashMap<String, Object>(), new String[0], null);
+        Request request = client.buildRequest(path, method, queryParams, new ArrayList<Pair>(), null, headers, new HashMap<String, Object>(), new String[0], null);
+        streamRequest(request, client, listener);
+    }
+
+    /** If we ever upgrade to okhttp 3...
```

Comment:
I think this might get interpreted as a javadoc with that extra `*`

Reply:
done.